pym/bob/cmds/build.py

```diff
@@ -939,24 +954,43 @@ def _cookPackageStep(self, packageStep, checkoutOnly, depth):
                 # empty. If the directory holds a result and was downloaded it
                 # we're done.
                 if BobState().getResultHash(prettyPackagePath) is None:
-                    audit = os.path.join(prettyPackagePath, "..", "audit.json.gz")
-                    if self.__archive.downloadPackage(packageBuildId, audit, prettyPackagePath, self.__verbose):
-                        self.__statistic.packagesDownloaded += 1
+                    if packageStep.isShared() and self.__sharedFolder and packageStep.isPackageRelocatable():
+                        if self._checkSharedPackage(asHexStr(packageBuildId)):
+                            sharedpackagepath = os.path.join(self.__sharedFolder, asHexStr(packageBuildId))
+                            print(colorize("     Link from shared location {}".format(sharedpackagepath), "32"))
```

Comment:
I would factor out the "link shared" boiler plate code into a dedicated function. The `_cookPackageStep` method is already barely readable...